OCPBUGS-19403: updates etcd procedure with OVN-K i/c (#64939)
openshift/openshift-docs: modules/dr-restoring-cluster-state.adoc

+
If the status is `Pending`, or the output lists more than one running etcd pod, wait a few minutes and check again.

. If you are using the `OVNKubernetes` network plugin, you must restart `ovnkube-controlplane` pods.
.. Delete all the `ovnkube-controlplane` pods by running the following command:
+
[source,terminal]
----
$ oc -n openshift-ovn-kubernetes delete pod -l app=ovnkube-control-plane
----
.. Verify all of the `ovnkube-controlplane` pods were redeployed by running the following command:
+
[source,terminal]
----
$ oc -n openshift-ovn-kubernetes get pod -l app=ovnkube-control-plane
----

> . If you are using the `OVNKubernetes` network plugin, restart the Open Virtual Network (OVN) Kubernetes pods on all the nodes one by one. Use the following steps to restart OVN-Kubernetes pods on each node:

Reviewer: Is there a reason that `OVNKubernetes` is in backticks, but "OVN-Kubernetes" is not? Are these different things?

Author: No just a mistake.

$ oc -n openshift-ovn-kubernetes get ds/ovnkube-master -o yaml | grep -E '<non-recovery_controller_ip_1>|<non-recovery_controller_ip_2>'
$ oc -n openshift-ovn-kubernetes get pod -l app=ovnkube-control-plane
----

. If you are using the `OVNKubernetes` network plugin, restart the Open Virtual Network (OVN) Kubernetes pods on all the nodes one by one. Use the following steps to restart OVN-Kubernetes pods on each node:
+
[NOTE]
[IMPORTANT]
====

> It can take at least 5-10 minutes for the OVN-Kubernetes control plane to be redeployed and the previous command to return empty output.

Author: @palonsoro does this need to be added back in as well?

Reviewer 2: I don't think so. The ovn-kubernetes pods are now much more lightweight. This sentence is present in 4.13 and older because CNO has to re-deploy the control plane single replica and it intentionally waits several minutes before doing so. That's no longer the case.

+
If the status is `Pending`, or the output lists more than one running etcd pod, wait a few minutes and check again.

. If you are using the `OVNKubernetes` network plugin, you must restart `ovnkube-controlplane` pods.

> .. Delete all the ovnkube-controlplane pods by running the following command:

Reviewer 3: Should `ovnkube-controlplane` be in backticks like the above line has it?  Same for line 291

+
[source,terminal]
----
$ oc -n openshift-ovn-kubernetes delete pod -l app=ovnkube-control-plane
----
.. Deploy all of the ovnkube-controlplane pods again by running the following command:
+
[source,terminal]
----

> $ oc -n openshift-ovn-kubernetes get pod -l app=ovnkube-control-plane

Reviewer 3: Definitely confirm with an SME, but I don't think this command will deploy the pods again. The `get` command just retrieves information.

Though actually now that I think about it, they are probably redeployed automatically because you deleted the pods in the previous command. So you might just need to reword the above step to not say "Deploy", but instead say something like verify that all of the pods are deployed again...

Reviewer 2: It should be `delete`, not `get`. Good catch.

Reviewer 3: @palonsoro The previous command (line 289) does already delete the pods.

This command looks like it might just be trying to verify that the pods were redeployed. So it might just be the wording above the command that needs to be changed (instead of saying "Deploy... " maybe it should say something about verifying that the pods were redeployed?)

Reviewer 2: Ok. I missed the context around. Then yes, you don't deploy but you wait for the pods to be deployed (by OpenShift)

----

> . If you are using OVNKubernetes network plugin, restart the Open Virtual Network (OVN) Kubernetes pods on all the nodes one by one. Steps to restart OVN-K pods in each node are as follows:

Reviewer 3: * To be consistent with how it's written above, do "If you are using the `OVNKubernetes` network plugin"
* Is "OVN-K" an approved shortened form, or should it be written out?
* Consider rewording the last sentence to be more active

```suggestion
. If you are using the `OVNKubernetes` network plugin, restart the Open Virtual Network (OVN) Kubernetes pods on all the nodes one by one. Use the following steps to restart OVN-K pods on each node:
```

+
[NOTE]
====

> Restart OVN-K pods in the following order: the recovery control plane host, the other control plane hosts (if available), and lastly the other nodes.

Reviewer 3: * Same question about OVN-K, should it be "OVN-Kubernetes"?
* Since this is a specific order, I'd consider making an actual ordered list

Reviewer 2: @JoeAldinger now that I re-read this: The paragraph that you are deleting with this change should not be deleted and you are deleting it. Please keep it.

====

.. Remove the northbound database (nbdb) and southbound database (sbdb). Access the recovery host and the remaining control plane nodes by using Secure Shell (SSH) and run the following command:
+
[source,terminal]
----
$ sudo rm -f /var/lib/ovn-ic/etc/*.db
----

.. Restart the OpenVSwitch services. Access the node by using Secure Shell (SSH) and run the following command:
+
[source,terminal]
----
$ sudo systemctl restart ovs-vswitchd ovsdb-server
----

> .. Delete the `ovnkube-node` pod running on the node by running the following command:

Reviewer 3: This uses "running" twice in the same sentence, with two different contexts. I would suggest rewording to avoid that. You can probably just remove the first instance of "running" and it would keep the same meaning

+
[source,terminal]
----
$ oc -n openshift-ovn-kubernetes delete pod -l app=ovnkube-node --field-selector=spec.nodeName==<node> <1>
----
+
<1> Replace `<node>` with the name of the node you are restarting.

> .. Verifty that the `ovnkube-node` pod is running again. Run the following command:

Reviewer 3: * Typo Verifty -> Verify
* I would combine the sentences "... by running the following command". However that does leave you in the same place as above with two "running"'s in there. Maybe you could reword though to avoid (that may have been why you broke it up in to two sentences, but there could be another way)

+
[source,terminal]
----
$ oc -n openshift-ovn-kubernetes get pod -l app=ovnkube-node --field-selector=spec.nodeName==<node>
----
+
[NOTE]
====

> It may take several minutes for the pods to restart.

Reviewer 3: ```suggestion
It might take several minutes for the pods to restart.
```

+
[source,terminal]
----
$ sudo rm -f /var/lib/ovn-ic/etc/*.db
----

.. Restart the OpenVSwitch services. Access the node by using Secure Shell (SSH) and run the following command:
+
[source,terminal]
----
$ sudo systemctl restart ovs-vswitchd ovsdb-server
----

.. Delete the `ovnkube-node` pod on the node by running the following command:
+
[source,terminal]
----

> $ oc -n openshift-ovn-kubernetes delete pod -l app=ovnkube-node --field-selector=spec.nodeName==<node> <1>

Reviewer: Is a callout necessary here? I feel like we can reword the instructions to avoid using a single callout in a single command with something like this:

```adoc
.. Delete the `ovnkube-node` pod on the node by running the following command, replacing `<node>` with the name of the node that you are restarting:
```

Author: I agree. More importantly I think this will create some issues with copying the code.
